pages/first-principles.markdown

```diff
@@ -106,6 +116,8 @@ On the third project, the cycle continues. You're constantly falling behind.
 
 The best option, in my experience, is for the organization to just accept deadlines don't work. That doesn't mean projects should take forever; practice Agile or MVP or what not.
 
+:speech_balloon: Hmm...it seems like the benefits of this principle come from (1) trusting that engineers will choose to make long term investments in the codebase and (2) providing enough autonomy that they can make those choices. If enough space is set aside for this, I don't see a problem with deadlines. In particular, Agile is all about making sprint commitments and hitting your definition of done by the end of the sprint. If those count as deadlines, we had plenty of them at the company where deadlines were rare. (Aside: can we not mention that company by name?) I will agree, however, that deadlines must originate from the people closest to the implementation, not be imposed on them.
```

Comment:
Sense of urgency is a great point, I need to work that in.

You are right that this may only apply to a narrow set of products/features, e.g. a basically mature SaaS app/company or something like that...

Per the sales/support training, I did make a point once to ask, ~six months ago, when the reason of "well, we need to align with sales/support on the release" was given for giving a date, I asked "okay, how long do they need?". The answer was "about two weeks". I was incredulous; I was expecting a quarter+, which would explain why alignment was so important.

Granted, this was just one data point, but it strengthens my musing that telling PM, sales, support, marketing to just not plan on pushing/preparing anything until what they consider v1 is *actually done*, is the best plan. E.g. if v1 is done on 1/1, and it takes them 1 month, or 2 months, or even 3 months to do the marketing/sales/etc. training, is that 3 months really that bad compared to telling them it will be ready 6 months ago, and then continually delaying/stalling/pressuring to make the date?

Granted, using a made up/extreme example, but that is my inclination. 

But, popping back, yeah good point another clarification of "within the contexts I've been in" is a good idea.